Cython/Compiler/MatchCaseNodes.py

```diff
@@ -182,15 +213,31 @@ class PatternNode(Node):
     times.
 
     as_targets   [NameNode]    any target assign by "as"
+
+    Generated in analysis:
+    comp_node   ExprNode     node to evaluate for the pattern
     """
 
-    child_attrs = ["as_targets"]
+    as_target = None
+    comp_node = None
+
+    # When pattern nodes are analysed it changes which children are important.
+    # Therefore have two different list of child_attrs and switch
+    initial_child_attrs = ["as_targets"]
+    post_analysis_child_attrs = ["comp_node"]
```

Comment:
Alternatively, we could list all attributes in `child_attrs` and set the initial-only attributes to None during analysis, maybe after moving them to backup attribute if they are still needed later. Would that fit the use case? It sounds less magic to me.

This could also be automated with an "initial only" or "pre analysis" or "clear during analysis" attribute names list that get all moved from `.original_name` to `._original_name` backup attributes by a suitable super class analysis method.